opsdroid/connector/matrix/connector.py

```diff
@@ -104,111 +109,139 @@ def _process_rooms_dict(self, rooms):
     @property
     def filter_json(self):
         """Define JSON filter to apply to incoming events."""
-        return {
-            "event_format": "client",
-            "account_data": {"limit": 0, "types": []},
-            "presence": {"limit": 0, "types": []},
-            "room": {"account_data": {"types": []}, "ephemeral": {"types": []}},
-        }
+        return json.dumps(
+            {
+                "event_format": "client",
+                "account_data": {"limit": 0, "types": []},
+                "presence": {"limit": 0, "types": []},
+                "room": {
+                    "account_data": {"types": []},
+                    "ephemeral": {"types": []},
+                    "state": {"types": []},
+                },
+            }
+        )
 
-    async def make_filter(self, api):
+    async def make_filter(self, api, fjson):
         """Make a filter on the server for future syncs."""
-        fjson = self.filter_json
-        resp = await api.create_filter(user_id=self.mxid, filter_params=fjson)
-        return resp["filter_id"]
+        path = f"/_matrix/client/r0/user/{self.mxid}/filter"
+        headers = {"Authorization": f"Bearer {api.token}"}
+        resp = await api.send(method="post", path=path, data=fjson, headers=headers)
+
+        resp_json = await resp.json()
+        return resp_json["filter_id"]
```

Comment:
I tested this branch out locally and this caused all my syncs to fail:
```suggestion
        return int(resp_json["filter_id"])
```